File: frontend/lib/api/activity.ts
import { apiClient } from "./client";

export interface ActivityLog {
  id: string;
  user_id: string;
  task_id: string | null;
  task_title: string | null;
  action: "task.created" | "task.updated" | "task.deleted";
  meta: Record<string, string> | null;
  created_at: string;
}

export const activityApi = {
  list: (limit = 20) =>
    apiClient.get<{ logs: ActivityLog[] }>(`/api/v1/activity?limit=${limit}`),
};

File: backend/internal/model/activity.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type ActivityLog struct {
	ID        uuid.UUID  `json:"id"`
	UserID    uuid.UUID  `json:"user_id"`
	TaskID    *uuid.UUID `json:"task_id"`
	TaskTitle *string    `json:"task_title"`
	Action    string     `json:"action"`
	Meta      any        `json:"meta,omitempty"`
	CreatedAt time.Time  `json:"created_at"`
}

const (
	ActionTaskCreated = "task.created"
	ActionTaskUpdated = "task.updated"
	ActionTaskDeleted = "task.deleted"
)
